datalad/interface/rerun.py

```diff
@@ -0,0 +1,300 @@
+                        err_info, status="error",
+                        message="branch name is required for orphan")
+                    return
+                else:
+                    ds.repo.checkout(revs[0] + "^", options=["--detach"])
+            else:
+                ds.repo.checkout(onto, options=["--detach"])
+        if do_checkout:
+            ds.repo.checkout("HEAD", ["-b", branch])
+
+        for rev in revs:
+            # pull run info out of the revision's commit message
+            try:
+                rec_msg, runinfo = get_commit_runinfo(ds.repo, rev)
+            except ValueError as exc:
+                yield dict(
+                    err_info, status='error',
+                    message=str(exc)
+                )
+                return
+            if not runinfo:
+                pick = False
+                try:
+                    ds.repo.repo.git.merge_base("--is-ancestor", rev, "HEAD")
+                except GitCommandError:  # Revision is NOT an ancestor of HEAD.
+                    pick = True
+
+                shortrev = ds.repo.repo.git.rev_parse("--short", rev)
+                err_msg = "no command for {} found; {}".format(
+                    shortrev,
+                    "cherry picking" if pick else "skipping")
+                yield dict(err_info, status='ok', message=err_msg)
+
+                if pick:
+                    ds.repo.repo.git.cherry_pick(rev)
+                continue
+
+            # now we have to find out what was modified during the
+            # last run, and enable re-modification ideally, we would
+            # bring back the entire state of the tree with #1424, but
+            # we limit ourself to file addition/not-in-place-modification
+            # for now
+            for r in ds.unlock(new_or_modified(ds, rev),
+                               return_type='generator', result_xfm=None):
+                yield r
+
+            for r in run_command(runinfo['cmd'], ds, rec_msg or message,
+                                 rerun_info=runinfo):
+                yield r
+
+
+def get_commit_runinfo(repo, commit="HEAD"):
+    """Return message and run record from a commit message
+
+    If none found - returns None, None; if anything goes wrong - throws
+    ValueError with the message describing the issue
+    """
+    commit_msg = repo.repo.git.show(commit, "--format=%s%n%n%b", "--no-patch")
+    cmdrun_regex = r'\[DATALAD RUNCMD\] (.*)=== Do not change lines below ' \
+                   r'===\n(.*)\n\^\^\^ Do not change lines above \^\^\^'
+    runinfo = re.match(cmdrun_regex, commit_msg,
+                       re.MULTILINE | re.DOTALL)
+    if not runinfo:
+        return None, None
+
+    rec_msg, runinfo = runinfo.groups()
+
+    try:
+        runinfo = json.loads(runinfo)
+    except Exception as e:
+        raise ValueError(
+            'cannot rerun command, command specification is not valid JSON: '
+            '%s' % str(e)
```

Comment:
I would recommend to use a helper `datalad.dochelpers.exc_str` instead of plain `str` for exceptions

Reply:
OK.  (I believe that code is just a direct copy from run.py's
--rerun.)